rust/rope/src/multiset.rs

```diff
@@ -180,6 +180,28 @@ impl Subset {
         sb.build()
     }
 
+    /// Compute the difference of two subsets. The count of an element in the
+    /// result is the subtraction of the counts of other from self.
+    pub fn subtract(&self, other: &Subset) -> Subset {
+        let mut sb = SubsetBuilder::new();
+        for zseg in self.zip(other) {
+            assert!(zseg.a_count >= zseg.b_count, "can't subtract {} from {}", zseg.a_count, zseg.b_count);
+            sb.push_segment(zseg.len, zseg.a_count - zseg.b_count);
+        }
+        sb.build()
+    }
+
+    /// Compute the bitwise xor of two subsets, useful as a symmetric
```

Comment:
This comment confused me a bit. In the case where the inputs are not multi (ie guaranteed to be 0 or 1), then this is a traditional symmetric difference. But afaict it's not primarily used in this context, deletes_from_union doesn't in general have this property. The explanation above (that it's a reversible difference) is good. I'm not sure it needs a full explanation about the difference between the multi and non-multi cases, but I think including the magic word "reversible" would help.